### ISSUE TITLE
Nested arrays are not parsed correctly

### ISSUE THREAD

**Issue by w-vi** [apiaryio/drafter#47]
The nested array object is not correctly parsed using drafter, see the example API Blueprint below.
In  Apiary I can see render only 

```json
{
  "categories": [
    {
      "name": "Category One",
      "subcategories": []
    }
  ]
}
```
instead of 
```json
{
  "categories": [
    {
      "name": "Category One",
      "subcategories":
      [
        {
          "name": "Sub category One"
        }
      ]
    }
  ]
}
```

Example Blueprint
```apib
FORMAT: 1A

# Test nested arrays-in-object-arrays

A navigation has multiple categories. Each category can have multiple subcategories.

# GET /navigation

+ Response 200 (application/json)

    + Attributes

        + categories (array)
            + (object)
                + name: Category One (string) - Name of the category
                + subcategories (array)
                    + (object)
                        + name: Sub category One (string) - Name of the subcategory
```

**EDIT by @pksunkara**: The MSON AST which is generated from the blueprint is wrong.

**Comment by JeroenVdb** [comment]
Related StackOverflow: http://stackoverflow.com/questions/30890933/nested-array-with-object-in-mson-is-empty-in-apiary-io-documentation/30893537

**Comment by pksunkara** [comment]
Thanks for the report @JeroenVdb. I am going to look into this now.

**Comment by pksunkara** [comment]
This is a bug in the parser itself. The generated MSON AST is not correct given the MSON.

**Comment by JeroenVdb** [comment]
Super @pksunkara can you give a shout here when this is fixed?
How can I see if this is then used by the apiary.io editor?

**Comment by pksunkara** [comment]
@JeroenVdb Apiary should have the bug fix as soon as we release the bug fix in here.

**Comment by alexturek** [comment]
@pksunkara Is [this](https://github.com/apiaryio/snowcrash/issues/330) the right snowcrash bug? Or is one filed?

**Comment by pksunkara** [comment]
@alexturek No, that is a different bug. This is a drafter or drafter.js bug which needs to be fixed.

**Comment by huiyiqun** [comment]
Please refer this: apiaryio/drafter.js#38

Working apib:
```apib
FORMAT: 1A

# Test nested arrays-in-object-arrays

A navigation has multiple categories. Each category can have multiple subcategories.

# GET /navigation

+ Response 200 (application/json)

    + Attributes

        + categories (array)
            + (object, fixed)
                + name: Category One (string) - Name of the category
                + subcategories (array)
                    + (object, fixed)
                        + name: Sub category One (string) - Name of the subcategory
```

OK, the schema works while the example not when I compiled it with aglio.

However, `fixed` means that there could not be any additional attributes in the nested objects, this may be a problem.

**Comment by obihann** [comment]
@pksunkara I mentioned this in a an apiary issue ( danielgtaylor/aglio#155), would you agree this is the same issue? 

Also, I'm more than happy to help out on this, just point me in the right direction. I know we spoke of this earlier but now that I am far more familiar with the drafter codebase I can probably understand the issue better.

**Comment by pksunkara** [comment]
@obihann No, this issue is completely different. The blueprint is not correctly being parsed at all in this issue while danielgtaylor/aglio#55 is about JSON Schema rendering.

**Comment by obihann** [comment]
@pksunkara strange, they seem so similar, oh well I'll trust your expertise :+1:. Once `nullable` has been properly implemented into all the tools I'll start looking into danielgtaylor/aglio#155 unless somebody has already.

**Comment by honzajavorek** [comment]
Just to verify, is this issue also about following problem or is it a new thing?

**Input:**

```
...
+ dimensions (array)
  + *20, 30, 40* (array[number], fixed)
...
```

**Generated JSON:**

```
{
  ...
  "dimensions": [],
  "20, 30, 40": [
    null
  ],
  ...
}
```

**Expected JSON:**

```
{
  ...
  "dimensions": [ [20, 30, 40] ],
  ...
}
```

**Comment by w-vi** [comment]
@honzajavorek it probably is the same issue or at least it is fixed in the #199 PR as is this one.

**Comment by w-vi** [comment]
@JeroenVdb It has been fixed a while ago can you verify?
